MRChemSoft/mrchem
# Refactor mrchem.in

All Python portions now live in the `python` subfolder of the source tree.

- Upgrade to latest `parselglossy`
- Regenerate the input parser
- Rewrite the CLI to use `argparse` (`optparse` is deprecated)
- Split the Python frontend into `mrchem.in`, a CLI module, and an API module. The latter two live in a `mrchem` Python module, of which the generated input parser is a submodule.
- All Python portions of the code (including `periodictable.py`) are now copied into the `python/mrchem` folder, that is: they are available as a Python module. ~**QUESTION** Should all of this be moved to some other folder in the source tree?~
- The Bohr to meter conversion factor is from CODATA 2018 *WARNING* This gives a different Angstrom to AU conversion factor that could impact results (though the tests are unaffected)
- Remove one Autocmake module in favor of better alternative in CMake>=3.12
- Update minimum CMake requirement to 3.12
- Clean up `.travis.yml` and `.circleci/config.yml`
- Remove outdated `lsdalton-guess` script and `lsdalton.py` module
